greymatter/inputs.cue

```diff
@@ -1,62 +1,93 @@
+	// team if you are not sure about this value. It's important that the zone
+	// matches the enterprise zone before deploying tenant services.
+	zone: "default-zone"
+	// Mesh name must match the enterprise mesh name so services appear
+	// in the greymatter dashboard.
+	name: "greymatter-mesh"
+	// Enabling Spire creates the necessary configurations to support sidecar
+	// to sidecar mutual TLS (mTLS) connections. The pre-requisite for this
+	// feature is that the enterprise team has deployed Spire or is using an
+	// external one. Therefore, enabling Spire on tenant sidecars should be
+	// done in accordance with enterprise-level Spire settings. The above mesh
+	// and zone names are used in Spire configuration and must match enterprise
+	// configurations.
+	enable_spire: false
+	// Sidecars publish metrics to Redis in support of sidecar/service
+	// health checks in the dashboard. A tenant does not need to change
+	// these configurations. Doing so may prevent operators from using
+	// health checks to troubleshoot sidecars/services.
+	redis: {
+		// The Redis key identifies the target for sidecar to Redis
+		// egress communication.
+		key: "redis"
+		// The Redis port indicates the localhost port from which egress
+		// requests flow out from the sidecar to Redis.
+		ingress_port: 10910
 	}
 }
 
-defaults: {
-	redis_cluster_name: "redis"
-
-	ports: {
-		default_ingress: 10808
-		edge_ingress:    10809
-		redis_ingress:   10910
-		metrics:         8081
-	}
+// A subset of a tenant's sidecar configurations can be modified in
+// isolation from the enterprise. These do not impact global sidecar
+// configurations.
+sidecars: {
+	// The ingress port that all tenant sidecars will listen on.
+	ingress_port: 10808
+}
 
-	edge: {
-		key:        "edge-example"
-		enable_tls: false
-		oidc: {
-			endpoint_host: ""
-			endpoint_port: 0
-			endpoint:      "https://\(endpoint_host):\(endpoint_port)"
-			domain:        ""
-			client_id:     "\(defaults.edge.key)"
-			client_secret: ""
-			realm:         ""
-			jwt_authn_provider: {
-				keycloak: {
-					issuer: "\(endpoint)/auth/realms/\(realm)"
-					audiences: ["\(defaults.edge.key)"]
-					local_jwks: {
-						inline_string: #"""
-					  {}
-					  """#
-					}
-					// If you want to use a remote JWKS provider, comment out local_jwks above, and 
-					// uncomment the below remote_jwks configuration. There are coinciding configurations
-					// in ./gm/outputs/edge.cue that you will also need to uncomment.
-					// remote_jwks: {
-					//  http_uri: {
-					//   uri:     "\(endpoint)/auth/realms/\(realm)/protocol/openid-connect/certs"
-					//   cluster: "edge_plus_to_keycloak" // this key should be unique across the mesh
-					//  }
-					// }
+// Tenant edge gateway for north/south traffic control. This is a separate
+// edge from the core greymatter edge. It is intended to provide tenants
+// with an edge gateway that has capabilities unique to the tenant.
+edge: {
```

Comment:
Once thing with this block, what happens when a tenant has to manage multiple edge nodes or bridges? What would this look like? Would they just define various `edge` blocks? I think if that's the case maybe we offer an `#edge` template in intermediates that they can reference here.

Reply:
Yeah, it's not a bad idea.

Reply:
Yes there will be one mesh edge but not for tenant repos, which is what these changes are focusing on.